tiled/adapters/arrow.py

```diff
@@ -0,0 +1,322 @@
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        partition: integer index of partition to be read.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        uri = self._partition_paths[partition]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in batches:
+                file_writer.write_batch(batch)
+            file_writer.close()
+
+    def write(
+        self,
+        data: Union[List[pyarrow.record_batch], pyarrow.record_batch, pandas.DataFrame],
+    ) -> None:
+        """
+        "Function to write the data as arrow format."
+        Parameters
+        ----------
+        data : data to write into arrow file. Can be a list of record batch, or pandas dataframe.
+        Returns
+        -------
+        """
+        if isinstance(data, pandas.DataFrame):
+            table = pyarrow.Table.from_pandas(data)
+            batches = table.to_batches()
+        else:
+            if not isinstance(data, list):
+                batches = [data]
+            else:
+                batches = data
+
+        schema = batches[0].schema
+
+        if self.structure().npartitions != 1:
+            raise NotImplementedError
+        uri = self._partition_paths[0]
+
+        with pyarrow.ipc.new_file(uri, schema) as file_writer:
+            for batch in data:
+                file_writer.write_batch(batch)
+            file_writer.close()
+
+    def read(self, *args: Any, **kwargs: Any) -> pandas.DataFrame:
+        """
+        The concatenated data from given set of partitions as pyarrow table.
+        Parameters
+        ----------
+        Returns
+        -------
+        Returns the concatenated pyarrow table as pandas dataframe.
+        """
+        if any(p is None for p in self.reader_handle):
+            raise ValueError("Not all partitions have been stored.")
+
+        data = pyarrow.concat_tables(
+            [partition.read_all() for partition in self.reader_handle]
+        )
+        return data.to_pandas()
+
+    def read_partition(
+        self,
+        partition: int,
+        fields: Optional[str] = None,
+    ) -> pandas.DataFrame:
+        """
+        Function to read a batch of data from a given partition.
+        Parameters
+        ----------
+        partition : the index of the partition to read.
+        fields : optional fields parameter.
+
+        Returns
+        -------
+        The pyarrow table corresponding to a given partition and batch as pandas dataframe.
+        """
+        df = self.reader_handle[partition]
```

Comment:
This opens all the files and then only uses one. Better to just open the file we need.